vernac/command.ml

```diff
@@ -203,7 +214,7 @@ match local with
   let () = maybe_declare_manual_implicits false gr imps in
   let () = Universes.register_universe_binders gr pl in
   let () = assumption_message ident in
-  let () = Typeclasses.declare_instance None false gr in
+  let () = if !axiom_into_instance then Typeclasses.declare_instance None false gr in
```

Comment:
Possibly we should ignore the option and always declare instance when Discharge (Variable outside sections).

Reply:
>Possibly we should ignore the option and always declare instance when Discharge (Variable outside sections).

Being able to take Variable/Context outside sections without breaking the instances is useful so yes.